src/ray/pubsub/subscriber.h

```diff
@@ -335,6 +349,14 @@ class Subscriber : public SubscriberInterface {
   const std::string subscriber_address_;
   const int subscriber_port_;
 
+  /// The command batch size for the subscriber.
+  const int64_t command_max_batch_size_;
+
+  /// Commands queue. Commands are reported in FIFO order to the publisher. This
+  /// guarantees the ordering of commands because they are delivered only by a single RPC
+  /// (long polling request).
+  std::queue<std::shared_ptr<rpc::Command>> commands_;
```

Comment:
Why shared pointer here? I think we push/pop only right? Maybe just rpc::Command and use std::move or emplace to avoid copying?

Reply:
yeah used shared pointer just to avoid copying the contents. Let me follow your suggestion.

Reply:
actually we want to use a pointer here to Swap (to avoid extra memory copy for grpc message). I. can change this to unique pointer if you'd like to, so let me know.

Reply:
or maybe I can Swap using (`&command`, but I am not sure if that's the common pattern )

Reply:
I am waiting for Yi's response.

Reply:
unique_ptr makes more sense here.
It depends on the implementation of rpc::Command, &command might also work.